pkg/controller/direct/bigquerybiglake/table_controller.go

```diff
@@ -0,0 +1,251 @@
+	gcpClient, err := m.client(ctx)
+	if err != nil {
+		return nil, err
+	}
+	return &TableAdapter{
+		id:        id,
+		gcpClient: gcpClient,
+		desired:   obj,
+	}, nil
+}
+
+func (m *modelTable) AdapterForURL(ctx context.Context, url string) (directbase.Adapter, error) {
+	// TODO: Support URLs
+	return nil, nil
+}
+
+type TableAdapter struct {
+	id        *krm.TableIdentity
+	gcpClient *gcp.MetastoreClient
+	desired   *krm.BigLakeTable
+	actual    *bigquerybiglakepb.Table
+}
+
+var _ directbase.Adapter = &TableAdapter{}
+
+// Find retrieves the GCP resource.
+// Return true means the object is found. This triggers Adapter `Update` call.
+// Return false means the object is not found. This triggers Adapter `Create` call.
+// Return a non-nil error requeues the requests.
+func (a *TableAdapter) Find(ctx context.Context) (bool, error) {
+	log := klog.FromContext(ctx)
+	log.V(2).Info("getting Table", "name", a.id)
+
+	req := &bigquerybiglakepb.GetTableRequest{Name: a.id.String()}
+	tablepb, err := a.gcpClient.GetTable(ctx, req)
+	if err != nil {
+		if direct.IsNotFound(err) {
+			return false, nil
+		}
+		return false, fmt.Errorf("getting Table %q: %w", a.id, err)
+	}
+
+	a.actual = tablepb
+	return true, nil
+}
+
+// Create creates the resource in GCP based on `spec` and update the Config Connector object `status` based on the GCP response.
+func (a *TableAdapter) Create(ctx context.Context, createOp *directbase.CreateOperation) error {
+	log := klog.FromContext(ctx)
+	log.V(2).Info("creating Table", "name", a.id)
+	mapCtx := &direct.MapContext{}
+
+	desired := a.desired.DeepCopy()
+	resource := BigLakeTableSpec_ToProto(mapCtx, &desired.Spec)
+	if mapCtx.Err() != nil {
+		return mapCtx.Err()
+	}
+
+	req := &bigquerybiglakepb.CreateTableRequest{
+		Parent:  a.id.Parent().String(),
+		Table:   resource,
+		TableId: a.id.ID(),
+	}
+	created, err := a.gcpClient.CreateTable(ctx, req)
+	if err != nil {
+		return fmt.Errorf("creating Table %s: %w", a.id, err)
+	}
+	log.V(2).Info("successfully created Table", "name", a.id)
+
+	status := &krm.BigLakeTableStatus{}
+	status.ObservedState = BigLakeTableObservedState_FromProto(mapCtx, created)
+	if mapCtx.Err() != nil {
+		return mapCtx.Err()
+	}
+	status.ExternalRef = direct.LazyPtr(a.id.String())
+	return createOp.UpdateStatus(ctx, status, nil)
+}
+
+// Update updates the resource in GCP based on `spec` and update the Config Connector object `status` based on the GCP response.
+func (a *TableAdapter) Update(ctx context.Context, updateOp *directbase.UpdateOperation) error {
+	log := klog.FromContext(ctx)
+	log.V(2).Info("updating Table", "name", a.id)
+	mapCtx := &direct.MapContext{}
+
+	desiredPb := BigLakeTableSpec_ToProto(mapCtx, &a.desired.DeepCopy().Spec)
+	if mapCtx.Err() != nil {
+		return mapCtx.Err()
+	}
+
+	paths := make(sets.Set[string])
+	{
+		var err error
+		paths, err = common.CompareProtoMessage(desiredPb, a.actual, common.BasicDiff)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
I like using "{}" to group logic as well, but it is a not recommended golang coding style. Thus I suggest removing line 170 and 176 to keep the code convention.